hooks/src/lib/utils/createFieldLengthMap.ts

```diff
@@ -0,0 +1,13 @@
+import { CardNumberFieldType } from '../config';
+
+export function createFieldLengthMap<T extends string>(
+  fields: CardNumberFieldType<T>[]
+): Record<T, number> {
+  return fields.reduce(
+    (acc, field) => {
+      acc[field.name] = field.length;
+      return acc;
+    },
+    {} as Record<T, number>
+  );
+}
```

Comment:
큰 차이는 없지만, reduce 에서 generic 을 쓸 수 있습니다. 

```suggestion
) {
  return fields.reduce<Record<T, number>>(
    (acc, field) => {
      acc[field.name] = field.length;
      return acc;
    },
    {} as Record<T, number>
  );
}
```

Reply:
그리고 이 함수가 잘못되지 않았지만요, fromEntries 를 써서 표현할 수도 있겠어요. 한번 고려해보아요~

Reply:
두 방법 모두 새로 알게된 지식이라 적용해보았습니다! 감사합니다 ㅎㅎ

[refactor: reduce에서 generic 쓰는 방향으로 변경](https://github.com/woowacourse/react-modules/pull/149/commits/b5c376661edb2f0264828edf77fd93ac0bb0dc5f)
[refactor: reduct를 fromEntries로 변경](https://github.com/woowacourse/react-modules/pull/149/commits/15644cac0fa7f8059efeaf8c090bfe712f185e6f)

Reply:
새로운 시도를 해주셨네요 ㅎㅎ 굿굿!!